### ISSUE TITLE
NullPointerException when writing indexed columns with null values

### ISSUE THREAD

**Issue by sadikovi** [sadikovi/riff#57]
Column filters fail to write null values.
Test to check:
```scala
test("write/read dataframe with null values + filter") {
  val implicits = spark.implicits
  withTempDir { dir =>
    import implicits._
    val df = Seq[(String, java.lang.Integer, java.lang.Long)](
      ("a", 1, 2L),
      (null, null, null),
      ("c", -1, 4L)
    ).toDF("col1", "col2", "col3").coalesce(1)

    df.write.option("index", "col1,col2,col3").riff(dir.toString / "table")
    val res = spark.read.riff(dir.toString / "table")

    checkAnswer(res.filter("col2 = 1"), df.filter("col2 = 1"))
    checkAnswer(res.filter("col2 = -1"), df.filter("col2 = -1"))
    checkAnswer(res.filter("col2 is null"), df.filter("col2 is null"))
    checkAnswer(res.filter("col3 = 2"), df.filter("col3 = 2"))
    checkAnswer(res.filter("col3 = 4"), df.filter("col3 = 4"))
    checkAnswer(res.filter("col3 is null"), df.filter("col3 is null"))
  }
}
```

**Comment by sadikovi** [comment]
It looks like this fails only for UTF8String, it is okay for integer and long.

**Comment by sadikovi** [comment]
Fixed in #65.
